Dynatrace/dynatrace-operator
# Replace cilium network policy and update proxy tests

# Description

Update cilium network policies to default network policies:
* Update documentation to no longer require cilium (replaced with istio)
* Remove todos for cilium

Update checks in istio/proxy/network tests to work:
* Remove curl pod manifest and create curl pod in code (most of the content was replaced anyway)
* Update proxy usage to `https_proxy` environment variable
* Update proxy availability check to reflect changes
  * webhook should show `CONNECT <webhook service>:443`
  * activegate should show `RUNNING`
* Update proxy annotations to disable injection to combined annotation

## How can this be tested?
Following e2e tests, that were previously skipped, should pass:
* `make tests/e2e/cloudnative/istio`
* `make tests/e2e/cloudnative/network`
* `make tests/e2e/cloudnative/proxy`
* `make tests/e2e/activegate/proxy`


## Checklist
